staging/src/k8s.io/client-go/plugin/pkg/client/auth/openstack/token.go

```diff
@@ -0,0 +1,123 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package openstack
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/gophercloud/gophercloud"
+	"github.com/gophercloud/gophercloud/openstack"
+	tokens2 "github.com/gophercloud/gophercloud/openstack/identity/v2/tokens"
+	tokens3 "github.com/gophercloud/gophercloud/openstack/identity/v3/tokens"
+	"github.com/gophercloud/gophercloud/openstack/utils"
+)
+
+const expiresAtDelta = 10 * time.Second
+
+type openstackToken struct {
+	ID        string
+	ExpiresAt time.Time
+}
```

Comment:
That's not a bad thought. From what I understand, there are two ways to use the existing Token type. Let me know if I'm missing another one.
1. Replace `openstackToken` with `Token`. If I do this, I won't be able to define the `Valid` method, because `Token` is in a different package. So I think this is not an option.
2. Embed `Token` in `openstackToken`. I can do this, but constructing a literal openstackToken becomes cumbersome:
```
&openstackToken{
		ID:        id,
		ExpiresAt: expiresAtTime,
	}
```
vs
```
&openstackToken{&tokens3.Token{
		ID:        id,
		ExpiresAt: expiresAtTime,
	}}
```